packages/@react-spectrum/tag/src/Tag.tsx

```diff
@@ -35,43 +35,43 @@ export function Tag<T>(props: SpectrumTagProps<T>) {
   let {styleProps} = useStyleProps(otherProps);
   let {hoverProps, isHovered} = useHover({});
   let {isFocused, isFocusVisible, focusProps} = useFocusRing({within: true});
-  let tagRef = useRef();
   let tagRowRef = useRef();
   let {clearButtonProps, labelProps, tagProps, tagRowProps} = useTag({
     ...props,
     isFocused,
     allowsRemoving,
     item,
     onRemove,
-    tagRef,
     tagRowRef
   }, state);
 
   return (
     <div
-      {...tagRowProps}
-      ref={tagRowRef}>
-      <div
-        {...mergeProps(tagProps, hoverProps, focusProps)}
-        className={classNames(
+      {...mergeProps(tagRowProps, hoverProps, focusProps)}
+      className={classNames(
           styles,
           'spectrum-Tags-item',
-          {
-            'focus-ring': isFocusVisible,
-            'is-focused': isFocused,
-            'is-hovered': isHovered
-          },
+        {
+          'focus-ring': isFocusVisible,
+          'is-focused': isFocused,
+          'is-hovered': isHovered,
+          'is-removable': allowsRemoving
+        },
           styleProps.className
         )}
-        ref={tagRef}>
+      ref={tagRowRef}>
+      <div
+        className={classNames(styles, 'spectrum-Tag-cell')}
+        {...tagProps}>
         <SlotProvider
           slots={{
             icon: {UNSAFE_className: classNames(styles, 'spectrum-Tag-icon'), size: 'XS'},
-            text: {UNSAFE_className: classNames(styles, 'spectrum-Tag-content', {'tags-removable': allowsRemoving}), ...labelProps}
+            text: {UNSAFE_className: classNames(styles, 'spectrum-Tag-content'), ...labelProps}
           }}>
-
           {typeof children === 'string' ? <Text>{children}</Text> : children}
-          {allowsRemoving && <TagRemoveButton item={item} {...clearButtonProps} UNSAFE_className={classNames(styles, 'spectrum-Tag-action')} />}
+          <ClearSlots>
+            {allowsRemoving && <TagRemoveButton item={item} {...clearButtonProps} UNSAFE_className={classNames(styles, 'spectrum-Tag-action')} />}
```

Comment:
Minor clean up nit: since this is no longer using slots to pass props to the `TagRemoveButton` (and in fact doesn't seem like it was before), mind removing the `useSlotProps` and `clearBtnRef` (not sure what that one is for) in `TagRemoveButton`